tests/integration/test_yarn.py

```diff
@@ -35,6 +35,20 @@
             marks=pytest.mark.xfail,  # temporary
             id="yarn_no_zero_installs",
         ),
```

Comment:
I think that if we're 100% sure that `yarn info` won't trigger any arbitrary code execution, we can set `plugins=[]` after `yarn info` was called, and just before we do any `yarn install`. Remember that the user can define any foo plugin that can handle a foo protocol, so we'd need to account for that scenario.

By a quick read on the yarn source code that you linked, I'm not 100% sure that this is the case, it seems there's a lot going on there. I'm cloning the repo now so I can better look at it locally.

Reply:
We discussed it some more over a private channel so I'm just summarizing our findings publicly for everyone to see:

- so far it looks like `plugins=[]` is really only for the user configured plugins, but not so much for the [default ones](https://v3.yarnpkg.com/features/plugins/#official-plugins) since even without this change the integration test provides expected results
- things change when deps include a custom locator based on a user-defined resolver plugin in which case there will be an error from Yarn about not being able to find a resolver
- parsing .yarnrc may be all we need to detect user-defined plugins (since it looks we can't affect the default ones much anyway) and hence produce a deterministic consistent error message by cachi2
- if we allow usage of plugins during the actual container build it will have consequences on the accuracy of the SBOM unless we make a note in there that a plugin (which we know nothing about) may have been executed during the build with unknown side-effects

Reply:
Ok, I found what the problem is. After this [PR](https://github.com/containerbuildsystem/cachi2/pull/399) came in, we started relying completely on the global cache. It seems that, after the dependency is cached, the original error no longer happens.

To reproduce it, try removing `~/.yarn/cache` and running cachi2 on the `no-available-resolver` branch on our test repo.

Reply:
It is still weird that calling `yarn info --all --recursive --cache` directly on the test repo does not cause the error. It is probably a combination of the yarn configuration we set on Cachi2. I'll try to figure this out tomorrow.

Reply:
> To reproduce it, try removing ~/.yarn/cache and running cachi2 

Doh! Right you are! Yeah, now I'm seeing the error with both cachi2 and plain yarn.

> It is still weird that calling yarn info --all --recursive --cache directly on the test repo does not cause the error

Have you dropped the plugins from `.yarnrc` explicitly? Because I can reproduce both ways, even with plain `yarn info`.

In any case, based on my previous post, since the idea is to disable all plugins for now and knowing that Yarn v3 would fail spectacularly without declaring some of them, I can see us proceeding in either way:

- fetch the `plugins` setting from `.yarnrc` ([default](https://v3.yarnpkg.com/api/modules/plugin_compat.html) plugins are loaded implicitly) and if **any explicit** plugin is declared, fail with a nice error (e.g. `Plugins unsupported`) OR
- still fetch the `plugins` setting, but detect if `exec` is declared (in v3 `exec` is [contrib](https://v3.yarnpkg.com/features/plugins#contrib-plugins) has to be imported, [v4 loads it by default](https://yarnpkg.com/api/plugin-exec)) and if no other plugin is requested, make sure we add it to the list of plugins before running `yarn info` (we'll fail with an unsupported protocol error later anyway)

Reply:
Or maybe we can just remove explicit plugins from .yarnrc, since it'll allow `yarn info` to pass and the request will fail before `yarn install` is called. 

Thoughts on our options here, @taylormadore?

Reply:
So I discussed with @taylormadore and we decided on keeping the default plugins in yarnrc.